flycheck.el

```diff
@@ -6330,6 +6331,118 @@ See URL `http://lesscss.org'."
           line-end))
   :modes less-css-mode)
 
+(defconst flycheck-luacheck-std-globals-list
+  '("_G" "lua51" "lua52" "lua52c" "lua53" "lua53c" "luajit" "min" "max" "none")
+  "List of standard globals.")
+
+(flycheck-def-option-var flycheck-luacheck-std-globals
+    (nth 0 flycheck-luacheck-std-globals-list)
+    luacheck
+  "Set standard globals.
+Must be one of the members in `flycheck-luacheck-std-globals-list'."
+  :type `(choice (const :tag "Globals of the Lua interpreter"
+                        ,(nth 0 flycheck-luacheck-std-globals-list))
+                 (const :tag "Globals of Lua 5.1"
+                        ,(nth 1 flycheck-luacheck-std-globals-list))
+                 (const :tag "Globals of Lua 5.2"
+                        ,(nth 2 flycheck-luacheck-std-globals-list))
+                 (const :tag "Globals of Lua 5.2 compiled with LUA_COMPAT_ALL"
+                        ,(nth 3 flycheck-luacheck-std-globals-list))
+                 (const :tag "Globals of Lua 5.3"
+                        ,(nth 4 flycheck-luacheck-std-globals-list))
+                 (const :tag "Globals of Lua 5.3 compiled with LUA_COMPAT_ALL"
+                        ,(nth 5 flycheck-luacheck-std-globals-list))
+                 (const :tag "Globals of LuaJIT 2.0"
+                        ,(nth 6 flycheck-luacheck-std-globals-list))
+                 (const :tag "Intersection of globals of Lua 5.1, 5.2 and LuaJIT 2.0"
+                        ,(nth 7 flycheck-luacheck-std-globals-list))
+                 (const :tag "Union of globals of Lua 5.1, 5.2 and LuaJIT 2.0"
+                        ,(nth 8 flycheck-luacheck-std-globals-list))
+                 (const :tag "No standard globals"
+                        ,(nth 9 flycheck-luacheck-std-globals-list)))
+  :safe (lambda (value) (member value flycheck-luacheck-std-globals-list))
```

Comment:
This is way too detailed and too complex. Please make this option a simple string.